File: package/android/src/main/java/com/audira/lib/reactnative/pingandroid/icmp/ICMPResult.kt
package com.audira.lib.reactnative.pingandroid.icmp

import com.facebook.react.bridge.ReadableMap

/**
 * @return A map that contain:
 * - `rtt`: `Number`
 * - `status`: `Number` - From the `PingConst.kt`
 * - `ttl`: `Number`
 * - `isEnded`: `Boolean`
 */
typealias ICMPResult = ReadableMap

File: package/src/ICMP/types/ICMPResult.ts
import {
	type Double,
} from 'react-native/Libraries/Types/CodegenTypes'

import {
	PingStatus,
} from '../../PingStatus'

export interface ICMPResult {
	rtt: Double,
	ttl: number,
	status: PingStatus,
	isEnded: boolean,
}
